dns/dnssec.py

```diff
@@ -482,6 +482,7 @@ def _sign(
     expiration: Optional[Union[datetime, str, int, float]] = None,
     lifetime: Optional[int] = None,
     verify: bool = False,
+    deterministic_signing: Optional[bool] = None,
```

Comment:
We might want to call this just `deterministic`

Reply:
Oh, and it should probably be a bool with a default; it doesn't need to be an Optional[bool] with default None as that just broadens the type signature.

Reply:
> Can you put this new arguemnt at the end? I know it's in an "_" function but it's still good to not break argument order in case anyone is monkey patching. "deterministic" is probably fine given we know we're signing.

Fixed in 5d59afee3d8625269ad6ee5d3649965e9e0e60e4

Reply:
> Oh, and it should probably be a bool with a default; it doesn't need to be an Optional[bool] with default None as that just broadens the type signature.

We can do just bool if accept that `True` means "generate deterministic signatures if possible". Would that be ok?

Reply:
I think that's reasonable, but would suggest adding something like `currently only ECDSA` or `currently, this only affects ECDSA` to the documentation.

Reply:
RSA and EDDSA are also deterministic :-)

Reply:
Yes, but the parameter doesn't affect them, because they're always deterministic.

Reply:
+1 to the "if possible" with clarifying doc plan

Reply:
Fixed in f51dbb3fe6a7994896fca760137fcf36a4465837 with clarification in 4e47c83a5e8bd77156728efa68af64ed858dfb77